### DISCUSSION TITLE
how to add part of all tables in one shard

### DISCUSSION THREAD

**Discussion by flyly0755** [sqlalchemy/sqlalchemy#7493] [Usage Questions]
for example, with tableA and tableB, I want add tableA and tableB in shard1, and add tableA only in shard2, 
it's easy to add all 2 tables in shard1 with command Base.metadata.create_all(bind=bind),
but how can i just add tableA in shard2?
i have a try with command 
Base.metadata.create_all(bind=bind, tables=tableA)
but failed with error:
clickhouse_sqlalchemy.exceptions.DatabaseException: Orig exception: Code: 60, e.displayText() = DB::Exception: Table stdfdb.tableB doesn't exist. (version 20.8.11.17 (official build))
talbeA and tableB both with Base attribute, whether first time with Base.metadata.create_all will always try to add both tables?
Then delete tableA, so you can use Base.metadata.create_all(bind=bind, tables=tableA) to just add tableA?

**Comment by zzzeek** [comment]
`metadata.create_all()`  as you noted includes a tables parameter, so you could limit the tables it creates using that list.    A more general method though is to use multiple `MetaData` collections.  This doesn't necessarily mean you need multiple declarative bases as you can provide subclasses of `Base` that have distinct metadata collections.  See the second example at https://docs.sqlalchemy.org/en/14/orm/declarative_config.html#abstract for this.

**Comment by zzzeek** [comment]
OK reading more closely, you have another problem:

```
Base.metadata.create_all(bind=bind, tables=tableA)
but failed with error:
clickhouse_sqlalchemy.exceptions.DatabaseException: Orig exception: Code: 60, e.displayText() = DB::Exception: Table stdfdb.tableB doesn't exist. (version 20.8.11.17 (official build))
```

I'm assuming you meant "tables=[tableA]".   Anyway, im not deeply familiar with "clickhouse" but it looks like your tableA has a dependency on tableB, such as a foreign key constraint.  so to have a database with just "tableA" you would need to alter the definition of tableA such that it has no server-side dependency on tableB, such as removing the ForeignKey directive.

**Reply by flyly0755** [reply]
yes, the correct way is "tables=[tableA]", tableA is not related with tableB. 
The real situation is that tableB is small table, data volume is not big, so i just need use shard1 to store data of tableB;
but tableA is big table, need more than one shard to stord data of tableB, so i neeed add tableA in every shard.
and use one distributed view of tableA to manage data of tableA.
